zerver/lib/create_user.py

```diff
@@ -117,13 +118,14 @@ def create_user_profile(
         onboarding_steps=orjson.dumps([]).decode(),
         default_language=default_language,
         delivery_email=email,
+        email_address_visibility=email_address_visibility,
         **extra_kwargs,
     )
     if bot_type or not active:
         password = None
     if user_profile.email_address_is_realm_public():
-        # If emails are visible to everyone, we can set this here and save a DB query
-        user_profile.email = get_display_email_address(user_profile)
+        # If emails are visible to everyone, we can set this here to the original email and save a DB query
+        user_profile.email = email
```

Comment:
I think `get_display_email_address` returned a value equal to `email` in this case, right? I think that might have been the motivation for calling the function here -- the small readability benefit of having us call the same function in either code path.

Reply:
Updated this to use `get_display_email_address`.